packages/@aws-cdk/aws-stepfunctions-tasks/lib/sns/publish.ts

```diff
@@ -98,8 +147,30 @@ export class SnsPublish extends sfn.TaskStateBase {
         TopicArn: this.props.topic.topicArn,
         Message: this.props.message.value,
         MessageStructure: this.props.messagePerSubscriptionType ? 'json' : undefined,
+        MessageAttributes: this._renderAttributes(),
         Subject: this.props.subject,
       }),
     };
   }
+  /**
+   * Provides the message attributes rendering
+   */
+  /**
+   * @internal
+   */
+  private _renderAttributes(): any {
+    if (!this.props.messageAttributes) {
+      return undefined;
+    }
+    const attrs: { [key: string]: any } = {};
+    for (const a of Object.keys(this.props.messageAttributes)) {
+      const attr = this.props.messageAttributes[a];
+      attrs[a] = {};
+      attrs[a].DataType = attr.type;
+      attrs[a][`${attr.type}Value`] = attr.value.value;
```

Comment:
I don't love the need for this function or this specific line. Is there a better solution?